src/main/java/org/testng/DependencyMap.java

```diff
@@ -61,11 +61,14 @@ public ITestNGMethod getMethodDependingOn(String methodName, ITestNGMethod fromM
     }
     for (ITestNGMethod m : l) {
       // If they are in the same class hierarchy, they must belong to the same instance,
-      // otherwise, it's a method depending on a method in a different class so we
-      // don't bother checking the instance
-      if (fromMethod.getRealClass().isAssignableFrom(m.getRealClass())) {
-        if (m.getInstance() == fromMethod.getInstance()) return m;
+      Object fromMethodInstance = fromMethod.getInstance();
+      Object mInstance = m.getInstance();
+      if (mInstance != null && fromMethodInstance != null &&
+              fromMethodInstance.getClass().isAssignableFrom(mInstance.getClass())) {
+        return m;
```

Comment:
Strange: it looks we are returning the same `m` in the if and else statement.

Reply:
Yes @juherr . Its kind of like the earlier logic. I have merely included the `Object` into the equation, which wasn't the case earlier. I haven't figured out an alternative for this yet.

Reply:
If both if and else are doing the same thing, we can remove the if and directly return the value ?

Reply:
@juherr - That doesn't work. I refactored it to make it a bit more explicit. I have amended the changes and pushed it in again. Please have a look.